src/main.rs

```diff
@@ -490,18 +549,26 @@ fn run_cli(cli: Cli) -> Result<()> {
                         (&RecordData::default()).into()
                     };
 
-                    let row = missing.insert_and_ref(&data, &remote_id)?;
-                    (row, data)
+                    let row = undefined.insert_and_ref(&data, &remote_id)?;
+                    (row, data, remote_id)
+                }
+                RemoteIdState::Null(_, _) => {
+                    unreachable!(
+                        "A local record should not be present in the `NullRecords` table."
+                    );
+                }
+                RemoteIdState::Invalid(_) => {
+                    unreachable!("All identifiers for local records should be valid.");
                 }
```

Comment:
I was going to check whether this is logically unreachable, but noticed that `RemoteIdState::Invalid` is never created (I could only find two consumptions, including this one).

Reply:
Good catch, this error category is only created when you call the `resolve` method on a `RecordId` and not when already passing a `RemoteId`